### ISSUE TITLE
pip cannot install jupyter ascending

### ISSUE THREAD

**Issue by ochalkley-advai** [untitled-ai/jupyter_ascending.vim#21]
When I follow the readme instructions I get the following error:

```
$ pip install jupyter_ascending
Collecting jupyter_ascending
  Using cached jupyter_ascending-0.1.27-py3-none-any.whl.metadata (743 bytes)
Collecting aiohttp<4.0.0,>=3.8.4 (from jupyter_ascending)
  Using cached aiohttp-3.9.5-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.5 kB)
Collecting editdistance<0.7.0,>=0.6.2 (from jupyter_ascending)
  Using cached editdistance-0.6.2.tar.gz (31 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [25 lines of output]
      Traceback (most recent call last):
        File "/home/oli/miniconda3/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/oli/miniconda3/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/oli/miniconda3/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-yeuod35o/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-yeuod35o/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-yeuod35o/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 25, in <module>
        File "/tmp/pip-build-env-yeuod35o/overlay/lib/python3.12/site-packages/Cython/Build/Dependencies.py", line 1010, in cythonize
          module_list, module_metadata = create_extension_list(
                                         ^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-yeuod35o/overlay/lib/python3.12/site-packages/Cython/Build/Dependencies.py", line 845, in create_extension_list
          for file in nonempty(sorted(extended_iglob(filepattern)), "'%s' doesn't match any files" % filepattern):
        File "/tmp/pip-build-env-yeuod35o/overlay/lib/python3.12/site-packages/Cython/Build/Dependencies.py", line 117, in nonempty
          raise ValueError(error_msg)
      ValueError: 'editdistance/bycython.pyx' doesn't match any files
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Comment by achrip** [comment]
I ran into the same problem. I'm using Windows 11, Miniconda3, and Python 3.12.2.

I assume that's an issue caused by `Cython 3.0` after going around the internet. Try using an older version of python, perhaps. I ended up using Python 3.9.2 and everything went smoothly.

**Comment by net-raider** [comment]
same, for me it worked with python<3.11

**Comment by juanMarinero** [comment]
The README is vague. It should state: 

```md
**NOT** compatible with Notebook 7. Use py<=3.11
```

Instead it points to [jupyter_ascending](https://github.com/imbue-ai/jupyter_ascending) which if one searchs finds its [#57](https://github.com/imbue-ai/jupyter_ascending/issues/57) which asks

> Support for Notebook 7 

TLDR. As @achrip and @net-raider suggested install a py version that not pip install Notebook 7.

---

I use 3.10.12. Further, if I afterwards try to update it `pip install --upgrade notebook` I get a clear ERROR message:

```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
jupyter-ascending 0.1.27 requires notebook<7.0.0,>=6.5.1, but you have notebook 7.4.2 which is incompatible.
Successfully installed notebook-7.4.2
```

I highlight it: **jupyter-ascending 0.1.27 requires notebook<7.0.0,>=6.5.1**

If then I try to `pip install --upgrade jupyter-ascending` it reverts the notebook version:

```
Installing collected packages: notebook
  Attempting uninstall: notebook
    Found existing installation: notebook 7.4.2
    Uninstalling notebook-7.4.2:
      Successfully uninstalled notebook-7.4.2
Successfully installed notebook-6.5.7
```

So a vicious circle.

If no further doubts please close the issue.


----
Bonus:

This was already solved in [stackoverflow/65918238](https://stackoverflow.com/questions/65918238/78589199#comment138552956_78589199). Quote from [jupyter_contrib_nbextensions](https://github.com/ipython-contrib/jupyter_contrib_nbextensions#jupyterlab):
> Due to major differences between the Jupyter Notebook and JupyterLab, the extensions in this repository will not work in JupyterLab
